Yank/yank.py

```diff
@@ -218,7 +218,7 @@ def resume(self, phases=None):
         # If no phases specified, construct a list of phases from the filename prefixes in the store directory.
         if phases is None:
             phases = utils.find_phases_in_store_directory(self._store_directory)
-        self._phases = phases.keys()
+        self._phases = sorted(phases.keys())  # sorting ensures all MPI processes run the same phase
```

Comment:
Alternatively, we could use `_mpicomm.bcast()` to sync these between processes, but this sounds like it should work fine.